### PR TITLE
Device volume db display

### PR DESCRIPTION
**Pull request by ffongmdsp**

### CODE REVIEW

File: src/app/services/player.service.js [hunk 1]
```diff
@@ -192,6 +198,14 @@ class PlayerService {
     }
   }
 
+  get remoteMiniDspShdVolumeDb() {
```
[the remark this answers is not in the document]
Comment (ffongmdsp) [reply src/app/services/player.service.js]: Noted. I made the changes and pushed. Please review.

File: src/app/services/player.service.js [hunk 2]
```diff
@@ -203,20 +217,42 @@ class PlayerService {
...
+        self.remoteVolume = self.lastVolumeUpdateValue;
+      },100, false, this);
+    }
+    else {
+      this.$log.warn(`set volume ${volume} same as last value - ignore.`);
     }
     this.localVolume = volume;
   }
 
+  get miniDspShdVolumeDbAvail() {
```
[the remark this answers is not in the document]
Comment (ffongmdsp) [reply src/app/services/player.service.js]: Noted. I made the changes and pushed. Please review.

File: src/app/themes/volumio3/components/volume-manager/volumio3-volume-manager.html [hunk 3]
```diff
@@ -9,6 +9,8 @@
       </div>
       <div class="bigText">{{volumeManager.playerService.volume}}</div>
       <div class="smallText">/100</div>
+      <div ng-if="volumeManager.playerService.miniDspShdVolumeDbAvail" class="volDbText">
```
[the remark this answers is not in the document]
Comment (ffongmdsp) [reply src/app/themes/volumio3/components/volume-manager/volumio3-volume-manager.html]: Noted. I made the changes and pushed. Please review.

File: src/app/themes/volumio3/components/volume-manager/volumio3-volume-manager.html [hunk 4]
```diff
@@ -86,14 +90,16 @@
       ng-class="{disabled: volumeManager.playerService.state.mute}"
       ng-model="volumeManager.playerService.volume"
       min="0"
-      step="5"
+      step="1"
```
[the remark this answers is not in the document]
Comment (ffongmdsp) [reply src/app/themes/volumio3/components/volume-manager/volumio3-volume-manager.html]: Noted. I made the changes and pushed. Please review. These are customization issues that we need to work out how to allow miniDSP to make these kind of changes to customers.

File: src/app/services/player.service.js [hunk 5]
```diff
@@ -203,20 +217,42 @@ class PlayerService {
...
   get volume(){
-    if(Date.now() - this.lastVolumeUpdateTime > 1000){
-      this.localVolume = this.remoteVolume;
-    }
+    this.localVolume = this.remoteVolume;
+    this._miniDspShdVolumeDb = this.remoteMiniDspShdVolumeDb;
     return this.localVolume;
   }
 
   set volume(volume){
```
[the remark this answers is not in the document]
Comment (ffongmdsp) [reply src/app/services/player.service.js]: Existing mechanism has a bug. That's why I made the change. I reverted the changes and pushed. Please review. These are customization issues that we need to work out how to allow miniDSP to make these kind of bug fix to customers.

**Comment by volumio** [comment]
Superseeded by https://github.com/volumio/Volumio2-UI/pull/798

**Pull request closed without merging**